src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java

```diff
@@ -0,0 +1,58 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+
+public class LineDrawerAdapter implements Job2dDriver {
+
+	public enum Line {
+		BASIC, DOTTED, SPECIAL
+	}
+
+	private int startX = 0, startY = 0;
+	private final DrawPanelController controller;
+	private Line lineType = Line.SPECIAL;
+
+	public LineDrawerAdapter(DrawPanelController controller){
+		this.controller = controller;
+	}
+
+	@Override
+	public void setPosition(int x, int y) {
+		this.startX = x;
+		this.startY = y;
+	}
+
+	public void setLineType(Line lineType) {
+		this.lineType = lineType;
+	}
+
+	public ILine getLineType() {
+		switch (lineType) {
+			case BASIC:
+				return LineFactory.getBasicLine();
+			case DOTTED:
+				return LineFactory.getDottedLine();
+			case SPECIAL:
+				return LineFactory.getSpecialLine();

```

Comment:
Unfortunately, this is not a generic solution. (e.g. new line type requires modifications in this class, unnecessary direct coupling with LineFactory)